experimental/PropertyDDS/packages/property-dds/src/propertyTree.ts

```diff
@@ -157,23 +159,31 @@ export class SharedPropertyTree extends SharedObject {
 		this._root.cleanDirty(BaseProperty.MODIFIED_STATE_FLAGS.DIRTY);
 	}
 
-	public get changeSet() {
+	public get changeSet(): SerializedChangeSet {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
 		return this.tipView;
 	}
+
+    public get activeCommit(): IPropertyTreeMessage {
+        if(this.localChanges.length > 1) {
```

Comment:
@ruiterr  I am not quite sure whether that is correct, if we start out we have an empty array e.g. no local changes, when we have local changes that are rebased we clear all but the first entry L746

Reply:
I think this is correct.

> we clear all but the first entry

I am not sure what is meant by this. The code in line 746 removes only the first entry, i.e. when we receive a commit and this commit is the first entry in the local changes, it is removed from the local changes (because it is now part of the remote changes).

Currently, we always have the local changes follow the remote changes, no branching. Therefore the last commit is always either the last commit of the local changes or if there are none the last in remote changes. In the  long run, we might again introduce a local and a remote branch, but those concepts don't exist at the moment.

Reply:
as with our offline discussion I changed the test for local changes